internal/model/datacenter.go

```diff
@@ -68,7 +68,8 @@ func (dc *Datacenter) FindHost(computeResourceReference mor) *mo.HostSystem {
 // GetResourcePool returns the name of the Resource Pool if is not the default
 func (dc *Datacenter) GetResourcePool(resourcePoolReference mor) (*mo.ResourcePool, bool) {
 	if !dc.IsDefaultResourcePool(resourcePoolReference) {
-		return dc.ResourcePools[resourcePoolReference], true
+		rp, ok := dc.ResourcePools[resourcePoolReference]
```

Comment:
IsDefaulResourcePool might return false because the ResourcePool does not exist, not only because it's not the default, so GetResourcePool could return true even though the ResourcePool does not exist